### PR TITLE
 Added solution for Project Euler problem 102

### PR DESCRIPTION
**Pull request by Cosine42**
### **Describe your change:**

[Project Euler 102](https://projecteuler.net/problem=102)

Added solution for the problem.

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.
#3823

### CODE REVIEW

**Comment by xcodz-dot** [comment]
fix:
* `pip install pre-commit`
* `pre-commit run -a` at root of repo (this might show it failed but it did not)

File: project_euler/problem_102/sol1.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+"""
+Project Euler Problem 102: https://projecteuler.net/problem=102 
+ 
+Problem statement: Three distinct points are plotted at random on a Cartesian plane, for which -1000 ≤ x, y ≤ 1000, such that a triangle is formed.
```
Comment (xcodz-dot) [inline project_euler/problem_102/sol1.py:4]:
```suggestion
Problem statement: Three distinct points are plotted at random 
on a Cartesian plane, for which -1000 ≤ x, y ≤ 1000, such that a triangle is formed.
```

File: project_euler/problem_102/sol1.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+Consider the following two triangles:
+
+A(-340,495), B(-153,-910), C(835,-947)
+X(-175,41), Y(-421,-714), Z(574,-645)
+
+It can be verified that triangle ABC contains the origin, whereas triangle XYZ does not.
+Using triangles.txt (right click and 'Save Link/Target As...'), a 27K text file containing the co-ordinates of one thousand "random" triangles, find the number of triangles for which the interior contains the origin.
```
Comment (xcodz-dot) [inline project_euler/problem_102/sol1.py:11]:
```suggestion
Using triangles.txt (right click and 'Save Link/Target As...'), a 27K text file containing
 the co-ordinates of one thousand "random" triangles, find the number of triangles 
 for which the interior contains the origin.
```

File: project_euler/problem_102/sol1.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+NOTE: The first two examples in the file represent the triangles in the example given above.
```
Comment (xcodz-dot) [inline project_euler/problem_102/sol1.py:12]:
```suggestion
NOTE: The first two examples in the file represent the triangles in
 the example given above.
```

**Review by xcodz-dot — CHANGES_REQUESTED** [review]
Shall fix flake8

**Comment by dhruvmanila** [comment]
@algorithms-keeper review

File: project_euler/problem_102/sol1.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+    >>> area()
+    690610.5
+
+    """
+
+    return abs((x1 * (y2 - y3) + x2 * (y3 - y1) + x3 * (y1 - y2)) / 2)
+
+
+def isInside(
```
Comment (xcodz-dot) [inline project_euler/problem_102/sol1.py:59]:
```suggestion
def is_inside(
```

File: project_euler/problem_102/sol1.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    >>> solution()
+    228
+
+    """
+    f = open(os.path.join(os.path.dirname(__file__), data_file))
+
+    count = 0
+    for line in f.readlines():
+
+        if isInside(*list(map(float, line.split(",")))):
```
Comment (xcodz-dot) [inline project_euler/problem_102/sol1.py:105]:
```suggestion
        if is_inside(*list(map(float, line.split(",")))):
```

**Review by xcodz-dot — CHANGES_REQUESTED** [review]
snake case naming convention

**Pull request closed without merging**